x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/SecurityTests.java

```diff
@@ -462,4 +482,36 @@ public void testValidateForFipsNoErrorsForDefaultSettings() {
         Security.validateForFips(settings);
         // no exception thrown
     }
+
+    public void testLicenseUpdateFailureHandlerUpdate() throws Exception {
+        Settings settings = Settings.builder().
+            put("xpack.security.authc.api_key.enabled", "true").
+            build();
+        Collection<Object> components = createComponentsWithSecurityNotExplicitlyEnabled(settings);
+        AuthenticationService service = findComponent(AuthenticationService.class, components);
+        assertNotNull(service);
+        RestRequest request = new FakeRestRequest();
+        final AtomicBoolean completed = new AtomicBoolean(false);
+        service.authenticate(request, ActionListener.wrap(result -> {assertTrue(completed.compareAndSet(false, true));
+                                                                    }, this::logAndFail));
+        assertTrue(completed.compareAndSet(true, false));
+        threadContext.stashContext();
+        licenseState.update(
+            randomFrom(License.OperationMode.GOLD, License.OperationMode.ENTERPRISE, License.OperationMode.PLATINUM),
+            true, null);
+        service.authenticate(request, ActionListener.wrap(result -> {assertTrue(completed.compareAndSet(false, true));
+                                                                    }, this::VerifyMissingAuthenticationException));
+        if(completed.get()){
+            fail("authentication succeeded but it shouldn't");
+        }
+    }
+
+    private void logAndFail(Exception e) {
+        logger.error("unexpected exception", e);
+        fail("unexpected exception " + e.getMessage());
+    }
+
+    private void VerifyMissingAuthenticationException(Exception e) {
+        assertThat(e.getMessage(), containsString("missing authentication credentials for REST request"));
```

Comment:
We need to verify the correct headers are set here. That is, that it includes a `WWW-Authenticate: Basic realm="security" charset="UTF-8"` header.

Reply:
Done